chains/ethereum/chain.go

```diff
@@ -1,6 +1,33 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: LGPL-3.0-only
+/*
+The ethereum package contains the logic for interacting with ethereum chains.
 
+There are 4 major components: The chain, the connection, the listener, and the writer.
+
+The Chain
+
+The chain handles the overheard and initalization of the chain. The `main` package calls the chain to create and connect to the proper ports and set the listener and writer.
+In addition to setting up the chain, it also has the functionality to start the chain and send transactions to the chain.
+
+The Connection
+
+The connection function within the chain is to poll blocks and pass those blocks to the listener.
```

Comment:
I have a vague idea of what polling actually is, going more detail would be very helpful